### PR TITLE
make it so .Do() can also follow redirects

### PR DESCRIPTION
**Pull request by brian-armstrong-discord**
this passes tests so i believe it is still working with the .Get()
family of functions. which leads me to believe it also will now work
with .Do(), given that they share a code path

### CODE REVIEW

File: client.go [hunk 1]
```diff
@@ -351,14 +347,45 @@ func (c *Client) DoDeadline(req *Request, resp *Response, deadline time.Time) er
...
+			err = errTooManyRedirects
+			break
+		}
+		location := resp.Header.peek(strLocation)
+		if len(location) == 0 {
+			err = errMissingLocation
+			break
+		}
+		url = getRedirectURL(url, location)
+	}
```
Comment (night) [inline client.go]: we need to follow redirects with cookies

Comment (brian-armstrong-discord) [reply client.go]: does this mean the original impl did it wrong then?

Comment (night) [reply client.go]: yup

Comment (brian-armstrong-discord) [reply client.go]: k, change added

**Review by night — APPROVED** [review]

**Pull request merged**